### DISCUSSION TITLE
Avoiding an email bomb/loop when adding custom email alerts.

### DISCUSSION THREAD

**Discussion by DebianGuru** [Security-Onion-Solutions/securityonion#10249] [2.4]
I've used the elastalert email feature to email me for certain alerts with no problem.

I needed to do some fancier stuff involving email, so I wrote a script that is called by the "command" alert type within Elastalert.  The script passes the alert contents (via stdin as provided by the command alert type) to ssmtp (small email client).

Here's where things go bad!  I have a sensor in the DMZ where our mail server is located.  When the SO box sends an alert to the mail server, the sensor detects the contents of the email and falsely identifies it as another alert, thus making a "loop" of alerts.  Within minutes you have hundreds of emails and I panic, running so-elastalert-stop!

I understand why this is happening, but why does that not happen when the regular "email" alert in Elastalert sends an email to the server.  The source (SO box) and mail server are the same, and both are sending SMTP to port 25.  I suppose I could setup a 1:1 NAT on my firewall (between SO box and DMZ) and do filtering on IPs, but I'd rather just find out why it only loops when using my custom script.  Any ideas?

**Comment by Acewiza** [comment]
Looks like everything is working exactly as expected.  Welcome to the world of false positives!  ;-)
